src/map.rs

```diff
@@ -266,29 +261,37 @@ impl<K: Key, V: Val<A>, A: Actor> Map<K, V, A> {
     }
 
     /// Update a value under some key, if the key is not present in the map,
-    /// the updater will be given the result of V::default().
-    pub fn update<F, I>(&self, key: I, ctx: AddCtx<A>, f: F) -> Op<K, V, A>
+    /// the updater will use the given `default` value.
+    pub fn update_with_default<F>(&self, key: K, ctx: AddCtx<A>, default: V, f: F) -> Op<K, V, A>
     where
         F: FnOnce(&V, AddCtx<A>) -> V::Op,
-        I: Into<K>,
     {
-        let key = key.into();
         let dot = ctx.dot.clone();
         let op = match self.entries.get(&key).map(|e| &e.val) {
             Some(data) => f(&data, ctx),
-            None => f(&V::default(), ctx),
+            None => f(&default, ctx),
         };
 
         Op::Up { dot, key, op }
     }
 
+    /// Update a value under some key, if the key is not present in the map,
+    /// the updater will be given the result of V::default().
+    pub fn update<F>(&self, key: K, ctx: AddCtx<A>, f: F) -> Op<K, V, A>
+    where
+        F: FnOnce(&V, AddCtx<A>) -> V::Op,
+        V: Default,
+    {
+        self.update_with_default(key, ctx, V::default(), f)
+    }
+
     /// Remove an entry from the Map
-    pub fn rm(&self, key: impl Into<K>, ctx: RmCtx<A>) -> Op<K, V, A> {
+    pub fn rm(&self, key: K, ctx: RmCtx<A>) -> Op<K, V, A> {
```

Comment:
so the `impl Into<K>` is nice if you are using string keys. It'll let you do things like `map.rm("foo", rm_ctx)` instead of `map.rm("foo".to_string(), rm_ctx)`. It should be essentially free for keys that are  in fact a `K`

Reply:
Ah, that's clever :sweat_smile: I'll revert and add docs. It confused me when I came across it so I think it's worth calling out.

> and the `impl Into<K>` was a nice usability feature that I'd like to see us keep (unless you have a good reason why we should remove it of course)

I don't think I can see any downsides :)